src/apps/bastion-protocol/helper/bastion-protocol.supply.token-helper.ts

```diff
@@ -0,0 +1,148 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { formatUnits } from 'ethers/lib/utils';
+import { compact } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { ETH_ADDR_ALIAS, ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import {
+  buildDollarDisplayItem,
+  buildPercentageDisplayItem
+} from '~app-toolkit/helpers/presentation/display-item.present';
+import { getTokenImg } from '~app-toolkit/helpers/presentation/image.present';
+import { ContractType } from '~position/contract.interface';
+import { BalanceDisplayMode } from '~position/display.interface';
+import { AppTokenPosition, ExchangeableAppTokenDataProps } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+import { BastionProtocolComptroller, BastionProtocolContractFactory, BastionProtocolCtoken } from '../contracts';
+
+export type BastionSupplyTokenDataProps = ExchangeableAppTokenDataProps & {
+  supplyApy: number;
+  borrowApy: number;
+  liquidity: number;
+  realmName: string;
+  comptrollerAddress: string;
+};
+
+type BastionSupplyTokenHelperParams<T = BastionProtocolComptroller, V = BastionProtocolCtoken> = {
+  comptrollerAddress: string;
+  realmName: string;
+  network: Network;
+  appId: string;
+  groupId: string;
+};
+
+@Injectable()
+export class BastionSupplyTokenHelper {
```

Comment:
Try to avoid creating this class entirely by using the `CompoundSupplyTokenHelper` from `src/apps/compound/helpers`. That helper should work for any Compound fork.

Reply:
You can take inspiration from `src/apps/market-xyz/avalanche/market-xyz.supply.token-fetcher.ts`, which is a lot like Bastion in that there are multiple isolated lending markets that all operate via separate comptrollers.

Reply:
Used `CompoundSupplyTokenHelper` now.

Reply:
Looks like our ABI is not compatible with compound's, I will fix that first.

Reply:
Because of the ABI difference, I think it's best to just stick with using our own helper.